server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -685,12 +685,19 @@ private Settings getOverrideSettingsInternal() {
                                 settingsBuilder,
                                 request.indexSettings(),
                                 clusterService.getSettings(),
-                                null
+                                null,
+                                clusterSettings
                             );
                         }
                         // remote store settings needs to be overridden if the remote store feature is enabled in the
                         // cluster where snapshot is being restored.
-                        MetadataCreateIndexService.updateRemoteStoreSettings(settingsBuilder, clusterService.getSettings());
+                        MetadataCreateIndexService.updateRemoteStoreSettings(
+                            settingsBuilder,
+                            clusterService.state(),
+                            clusterSettings,
+                            clusterService.getSettings(),
+                            request.getDescription()
+                        );
```

Comment:
can we add an IT for same ?

Reply:
Added here `RemoteStoreMigrationSettingsUpdateIT` > `testNewRestoredIndexIsRemoteStoreBackedForRemoteStoreDirectionAndMixedMode`